whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -0,0 +1,234 @@
+    public static final List<SortBuilder<?>> SORT_BUILDERS = Arrays.asList(SortBuilders.scoreSort(), SortBuilders.fieldSort("lookup-key.keyword").unmappedType("string"));
+
+    private final FullTextIndex fullTextIndex;
+    private final AccessControlListManager accessControlListManager;
+    private final ElasticIndexService elasticIndexService;
+
+    private final Source source;
+    private final RpslObjectDao objectDao;
+
+    private final int maxResultSize;
+
+    @Autowired
+    public ElasticFulltextSearch(final FullTextIndex fullTextIndex,
+                                 final ElasticIndexService elasticIndexService,
+                                 @Qualifier("jdbcRpslObjectSlaveDao") final RpslObjectDao rpslObjectDao,
+                                 final AccessControlListManager accessControlListManager,
+                                 final SourceContext sourceContext,
+                                 final ApplicationVersion applicationVersion,
+                                 @Value("${fulltext.search.max.results:100}") final int maxResultSize) {
+        super(applicationVersion);
+
+        this.fullTextIndex = fullTextIndex;
+        this.accessControlListManager = accessControlListManager;
+        this.source = sourceContext.getCurrentSource();
+        this.objectDao = rpslObjectDao;
+        this.maxResultSize = maxResultSize;
+        this.elasticIndexService = elasticIndexService;
+    }
+
+    @Override
+    public SearchResponse performSearch(final SearchRequest searchRequest, final String remoteAddr) throws IOException {
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+
+        if (searchRequest.getStart() > maxResultSize) {
+            throw new IllegalArgumentException("Too many rows");
+        }
+
+        return new ElasticSearchAccountingCallback<SearchResponse>(accessControlListManager, remoteAddr, source) {
+
+            @Override
+            protected SearchResponse doSearch() throws IOException {
+
+                final int start = Math.max(0, searchRequest.getStart());
+                final HighlightBuilder highlightBuilder = new HighlightBuilder()
+                        .postTags(getHighlightTag(searchRequest.getFormat(), searchRequest.getHighlightPost()))
+                        .preTags(getHighlightTag(searchRequest.getFormat(), searchRequest.getHighlightPre()))
+                        .field("*");
+
+                final SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+                        .query(getQueryBuilder(searchRequest.getQuery()))
+                        .size(searchRequest.getRows()).from(start)
+                        .aggregation(AGGREGATION_BUILDER)
+                        .sort(SORT_BUILDERS)
+                        .highlighter(highlightBuilder);
+
+                final org.elasticsearch.action.search.SearchRequest fulltextRequest = new org.elasticsearch.action.search.SearchRequest(elasticIndexService.getWHOIS_INDEX());
```

Comment:
Remove package from class (there's only one SearchRequest class?)

Reply:
It als exists in package : net.ripe.db.whois.api.fulltextsearch

Reply:
OK, if there's a duplicate class you can leave the package path